app/src/main/java/sdp/moneyrun/ui/menu/MenuActivity.java

```diff
@@ -248,49 +204,90 @@ private void setNavigationViewListener() {
         navigationView.setNavigationItemSelectedListener(this);
     }
 
-    private void setPutExtraArguments(Intent intent){
+    private void setPutExtraArguments(Intent intent) {
         intent.putExtra("user", user);
     }
 
-    public void onButtonSwitchToActivity(Class activityClass, boolean shouldFinish){
+    public void onButtonSwitchToActivity(Class activityClass, boolean shouldFinish) {
         Intent switchActivity = new Intent(MenuActivity.this, activityClass);
         setPutExtraArguments(switchActivity);
         startActivity(switchActivity);
-        if(shouldFinish){
+        if (shouldFinish) {
             finish();
         }
     }
 
-    public void setGuestPlayerFields(boolean guest){
-        if(guest){
+    public void setGuestPlayerFields(boolean guest) {
+        if (guest) {
             Button joinGame = findViewById(R.id.join_game);
             joinGame.setEnabled(false);
         }
     }
 
+    /////////////////////////////////////////////////////WEATHER IMPLEMENTATION
```

Comment:
Could you put these  underneath the class declaration? Also put the locationListener inside the function